core/src/upgrade/apply.rs

```diff
@@ -137,22 +141,30 @@ where
                             return Poll::Pending;
                         }
                     };
+                    let name = SmallVec::from_slice(info.protocol_name());
                     self.inner = InboundUpgradeApplyState::Upgrade {
                         future: Box::pin(upgrade.upgrade_inbound(io, info.0)),
+                        name,
                     };
                 }
-                InboundUpgradeApplyState::Upgrade { mut future } => {
+                InboundUpgradeApplyState::Upgrade { mut future, name } => {
                     match Future::poll(Pin::new(&mut future), cx) {
                         Poll::Pending => {
-                            self.inner = InboundUpgradeApplyState::Upgrade { future };
+                            self.inner = InboundUpgradeApplyState::Upgrade { future, name };
                             return Poll::Pending;
                         }
                         Poll::Ready(Ok(x)) => {
-                            debug!("Successfully applied negotiated protocol");
+                            log::trace!(
+                                "Successfully applied negotiated inbound protocol {}",
```

Comment:
```suggestion
                                "Upgraded inbound stream to {}",
```

Perhaps we can shorted this a bit?